pkg/storage/tsdb/block/meta.go

```diff
@@ -47,6 +47,11 @@ type Meta struct {
 	Thanos ThanosMeta `json:"thanos"`
 }
 
+// IsCompacted returns whether the block has been compacted yet.
+func (m *Meta) IsCompacted() bool {
```

Comment:
[food for thought] In Prometheus terminology, any block is a compacted block. Even level 1 blocks are compacted, because they're compacted from the TSDB Head. I find a bit confusing calling it "compacted" here and in the rest of the code.

Reply:
A better way to handle may be tracking the number of queries blocks by compaction level (see `BlockMetaCompaction` -> `Level`).

Reply:
Compaction level would seem more appropriate, but unfortunately compaction level alone doesn't distinguish between recently uploaded blocks from ingesters (level 1) with backfilled blocks (also level 1). The overall goal for the metric here is to determine which blocks have never been compacted by a compactor (it might be better to call this metric non_compacted_blocks...). Curious if you still think reporting level is appropriate?

Reply:
Track the metric by source and compaction level (but not on a per-tenant basis, otherwise cardinality explodes). I still prefer reporting on compaction level cause can also be a signal of how much the compactor is lagging behind.

Reply:
Also just checking the source is not enough to know if it was actually compacted. Due to the split and merge compactor, you may still query split blocks (whose source is compactor) which have not been compacted yet.

Reply:
I think tracking by source and level will not be more complicated than what you already did. You can just change `queryStats.nonCompactedBlocksQueried` into a `blocksQueryedBySourceAndLevel map[blockSourceAndLevel]int`

where `blockSourceAndLevel` is something like this:

```
type blockSourceAndLevel struct {
  source string
  level int
}
```

Reply:
If you do decide to include the level in the metric, be aware that we already have `cortex_bucket_store_blocks_loaded_by_duration{duration="2h"}`. I think it would be nice to attempt to retain some consistency between metrics and also export the level as a duration (`2h`) instead of an integer (`1`) in this metric.

Alternatively, we can use compaction level as an integer and remove `cortex_bucket_store_blocks_loaded_by_duration` because the existing metric was used to gauge how much the compactor is falling behind https://github.com/grafana/mimir/pull/6074

I think I slightly prefer the second approach, but don't really mind either way.

Reply:
I realized that `cortex_bucket_store_blocks_loaded_by_duration` is actually all the blocks that are available via the store-gateway - [code](https://github.com/grafana/mimir/blob/71ff092321b2d065dd6e0ed342e24a4384341ad1/pkg/storegateway/bucket.go#L308-L330). @jhalterman are you interested particularly in what's being served with each query or what's in scope of the store-gateway? We might be able to extend the [existing metric](https://github.com/grafana/mimir/blob/031381eed5299cf457f80bfdcb9d1b6ff4e53ade/pkg/storegateway/bucket_stores.go#L552-L568) in the interest of lower cardinality

Reply:
I adopted the blockAndSourceLevel approach, since viewing things by level seems useful. @dimitarvdimitrov - perhaps cortex_bucket_store_blocks_loaded_by_duration could be removed in a separate PR if it's not needed or redundant with this?

Reply:
> Any input on the metric name and whether it should be a summary or counter?

Since you're going with the level and source of currently queried blocks I think a counter makes sense.

Reply:
Marking resolved, as this now uses the summary metric `cortex_bucket_store_series_blocks_queried` with source and level fields.